### PR TITLE
fixes #15321

### PR DESCRIPTION
**Pull request by ar45**
Models._get_unique_checks() does not
return grandparents' unique checks

### CODE REVIEW

**Comment by charettes** [comment]
Hi @ar45, could you add a test for this?

**Comment by ar45** [comment]
I added the tests

**Comment by charettes** [comment]
Buildbot, test this please.

File: tests/model_inheritance/models.py [hunk 1]
```diff
@@ -186,3 +186,31 @@ class Base(models.Model):
...
+class GrandParent(models.Model):
+    first_name = models.CharField(max_length=80)
+    last_name = models.CharField(max_length=80)
+    email = models.EmailField(unique=True)
+
+    class Meta:
+        unique_together = ('first_name', 'last_name')
+
+    def __str__(self):
+        return '"%s %s" <%s>' % (self.first_name, self.last_name, self.email)
```
Comment (charettes) [inline tests/model_inheritance/models.py]: Is this method used anywhere in the tests?

Comment (ar45) [inline tests/model_inheritance/models.py]: oh, its not used.

**Comment by charettes** [comment]
Created #4045 with the following modifications:

1. Altered commit message to respect the documented format (`Fixed #XXXX -- ` followed by a summary in the past tense).
2. Removed unused `__str__` method`.
3. Used `setUpTestData` and avoid saving a `GrandChild` instance to the database.

If the build passes I will proceed to commit and backport to `stable/1.8.x`.

**Pull request closed without merging**